src/printer.js

```diff
@@ -3831,7 +3831,7 @@ function nodeStr(node, options) {
   const alternate = preferred === single ? double : single;
 
   let shouldUseAlternateQuote = false;
-  let canChangeDirectiveQuotes = true;
+  let canChangeDirectiveQuotes = !rawContent.includes('\\');
```

Comment:
Sure, it was the cases with escaped characters in the literals. Adding `--no-semi` makes it a little easier to see, so here's the full output of `prettier tests/quotes/strings.js --debug-check --no-semi`:

<details>

```
tests/quotes/strings.js
ast(input) !== ast(prettier(input))
Index: 
===================================================================
--- 	
+++ 	
@@ -73,5 +73,5 @@
         "value": {
           "type": "DirectiveLiteral",
-          "value": "\\'"
+          "value": "'"
         }
       },
@@ -80,5 +80,5 @@
         "value": {
           "type": "DirectiveLiteral",
-          "value": "\\'"
+          "value": "'"
         }
       },

Index: 
===================================================================
--- 	
+++ 	
@@ -9,15 +9,15 @@
 // Simple strings.
 "abc"
-'abc'
+"abc"
 
 // Escape.
-'\0'
+"\0"
 
 // Emoji.
-'🐶'
+"🐶"
 
 // Empty string.
 ""
-''
+""
 
 // Single double quote.
@@ -27,8 +27,8 @@
 // Single single quote.
 "'"
-'\''
+"'"
 
 // Unnecessary escapes.
-"\'"
+"'"
 '\"'
 
@@ -54,6 +54,6 @@
 
 // Single backslash.
-'\\'
 "\\"
+"\\"
 
 // Backslases.


```
</details>